src/room.cpp

```diff
@@ -37,8 +37,6 @@ void Room::Destroy() {
         }
         server = nullptr;
     }
-
-    room_thread->join();
```

Comment:
Not sure about this change, I guess it's fine, but I feel like we should continue handling packets even during shutdown.
Thread closure (setting state + joining thread) should probably happen after shutdown, before destroying the instance.

I'll add an issue once we merged this